### PR TITLE
[DOCS] Fix links to ES upgrade docs

### PR DESCRIPTION
**Pull request by jrodewig**
https://github.com/elastic/elasticsearch/pull/83489 reorganized the ES upgrade docs for 8.0+. This updates a few links that changed.

Relates to https://github.com/elastic/docs/pull/2312.

### CODE REVIEW

File: docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc [hunk 1]
```diff
@@ -127,7 +127,7 @@ ECK migrates data away from the Elasticsearch nodes due to be removed and then d
...
 +
-ECK performs a rolling upgrade of the corresponding Elasticsearch nodes. It follows the link:https://www.elastic.co/guide/en/elasticsearch/reference/current/rolling-upgrades.html[Elasticsearch rolling upgrade best practices] to update the underlying Pods while maintaining the availability of the Elasticsearch cluster where possible. In most cases, the process simply involves restarting Elasticsearch nodes one-by-one. Note that some cluster topologies may be impossible to deploy without making the cluster unavailable (check <<{p}-orchestration-limitations>> ).
+ECK performs a rolling upgrade of the corresponding Elasticsearch nodes. It follows the link:{stack-ref}/upgrading-elasticsearch.html#rolling-upgrades[Elasticsearch rolling upgrade best practices] to update the underlying Pods while maintaining the availability of the Elasticsearch cluster where possible. In most cases, the process simply involves restarting Elasticsearch nodes one-by-one. Note that some cluster topologies may be impossible to deploy without making the cluster unavailable (check <<{p}-orchestration-limitations>> ).
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:130]:
```suggestion
ECK performs a rolling upgrade of the corresponding Elasticsearch nodes. It follows the link:{stack-ref}/upgrading-elasticsearch.html[Elasticsearch rolling upgrade best practices] to update the underlying Pods while maintaining the availability of the Elasticsearch cluster where possible. In most cases, the process simply involves restarting Elasticsearch nodes one-by-one. Note that some cluster topologies may be impossible to deploy without making the cluster unavailable (check <<{p}-orchestration-limitations>> ).
```

Looking at the new page structure I think the main section (above the `#rolling-upgrades` anchor) is what is most relevant here.

**Review by pebrc — APPROVED** [review]

**Comment by pebrc** [comment]
@jrodewig are you going to raise the backport PRs? If not I can do it next week.

**Comment by jrodewig** [comment]
Thanks @pebrc. I'll raise the backport PRs, but I'll need you, @alaudazzi, or someone with permissions to merge them in.

**Comment by pebrc** [comment]
@jrodewig sure no problem!

**Pull request merged**